R/bambu-processReads_utilityCreateJunctionTables.R

```diff
@@ -0,0 +1,260 @@
+                                          genomeSequence, stranded = FALSE, 
+                                          verbose = FALSE) {
+  start.ptm <- proc.time()
+  #summarise junction counts and strand for all reads
+  uniqueJunctions <- createJunctionTable(unlisted_junctions,
+                                         genomeSequence = genomeSequence)
+  end.ptm <- proc.time()
+  if (verbose) message("Finished creating junction list with splice motif
+        in ", round((end.ptm - start.ptm)[3] / 60, 1), " mins.")
+  
+  uniqueAnnotatedIntrons <- unique(unlistIntrons(annotations, 
+                                                 use.ids = FALSE))
+  # correct strand of junctions based on (inferred) strand of reads
+  strand(uniqueJunctions) <- junctionStrandCorrection(uniqueJunctions,
+                                                      unlisted_junctions, 
+                                                      uniqueAnnotatedIntrons,
+                                                      stranded = stranded, 
+                                                      verbose = verbose)
+  # add annotation labels to junctions
+  mcols(uniqueJunctions) <- tibble(as.data.frame(uniqueJunctions)) %>% 
+    mutate(annotatedJunction = (!is.na(GenomicRanges::match(uniqueJunctions,
+                                                            uniqueAnnotatedIntrons)))) %>% 
+    group_by(seqnames) %>% 
+    mutate(annotatedStart = start %in% start[annotatedJunction],
+           annotatedEnd = end %in% end[annotatedJunction]) %>% ungroup() %>%
+    select(score, spliceMotif, spliceStrand, junctionStartName, junctionEndName,
+           startScore, endScore, id, annotatedJunction, annotatedStart, annotatedEnd)
+  # correct junction coordinates using logistic regression classifier
+  uniqueJunctions <- junctionErrorCorrection(uniqueJunctions, verbose)
+  return(uniqueJunctions)
+}
+
+
+
+#' Get unlisted intron ranges from exon ranges list
+#' @noRd
+unlistIntrons <- function(x, use.ids = TRUE, use.names = FALSE) {
+  # License note: This function is adopted from the GenomicAlignments 
+  # package (Author: Hervé Pagès, Valerie Obenchain, Martin Morgan)
+  # License Artistic-2.0
+  # https://doi.org/doi:10.18129/B9.bioc.GenomicAlignments
+  
+  flat <- unlist(x, use.names = FALSE)
+  gaps <- gaps(ranges(x))
+  
+  firstseg <- start(PartitioningByWidth(x))
+  seqnms <- rep(seqnames(flat)[firstseg], elementNROWS(gaps))
+  strand <- rep(strand(flat)[firstseg], elementNROWS(gaps))
+  
+  gr <- GenomicRanges::GRanges(seqnms, unlist(gaps,
+                                              use.names = use.names), strand)
+  if (use.ids & !is.null(mcols(x, use.names = FALSE)$id)) 
+    mcols(gr)$id <- rep(mcols(x)$id, elementNROWS(gaps))
+  return(gr)
+}
+
+
+
+#' Create Junction tables from unlisted junction granges
+#' @importFrom BiocParallel bppram bpvec
+#' @noRd
+createJunctionTable <- function(unlisted_junctions,
```

Comment:
This function also has a length slightly over 50 lines?

Reply:
I removed a check for constent seqlevels which was redundant, the function is now below 50 lines